persistence/state.go

```diff
@@ -17,12 +17,11 @@ import (
 type merkleTree float64
 
 type stateTrees struct {
-	merkleTrees map[merkleTree]*smt.SparseMerkleTree
+	merkleTrees map[merkleTree]*smt.SMT
 
-	// nodeStores & valueStore are part of the SMT, but references are kept below for convenience
+	// nodeStores are part of the SMT, but references are kept below for convenience
 	// and debugging purposes
-	nodeStores  map[merkleTree]kvstore.KVStore
-	valueStores map[merkleTree]kvstore.KVStore
```

Comment:
Were the value stores even used for anything? I considered a similar removal for my refactor but hadn't done it yet just because I was more worried about the rest of the functionality.

Reply:
Yes in the old version they were being used I haven't read through the non-lazy loading code much but they were required.